ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/clientlibs/editor/utils/datamodel.js

```diff
@@ -0,0 +1,248 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2024 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+(function($, channel, Coral) {
+    "use strict";
+
+    window.CQ = window.CQ || {};
+    window.CQ.FormsCoreComponents = window.CQ.FormsCoreComponents || {};
+    window.CQ.FormsCoreComponents.Utils = window.CQ.FormsCoreComponents.Utils || {};
+    const DataModel = window.CQ.FormsCoreComponents.Utils.DataModel = window.CQ.FormsCoreComponents.Utils.DataModel || {};
+
+    var FORM_CONTAINER_SELECTOR = "[data-cmp-is='adaptiveFormContainer']";
```

Comment:
None of this code is going to be changed by core component developers, let's move this code to cq-guides (like we have done for replace / rule editor etc), since this logic is very specific

Reply:
Yes, these codes should not require any customization. In that case Data Model code will also required to moved to avoid duplicacy right ?

Reply:
Yes, I agree that the data model code shouldn't be included here. We should keep the codebase clean by only including the relevant components that are customizable or useful for developers